src/utils/parseError.ts

```diff
@@ -162,20 +163,41 @@ export const parseError = async (
 
   switch (e.code) {
     case EthersErrorType.CallExecption:
-      if (e.reason?.includes?.includes(EthersErrorMessage.ERC20Allowance)) {
-        return new TransactionError(
-          LifiErrorCode.AllowanceRequired,
+      const defaultErrorMessage = await getTransactionNotSentMessage(
+        step,
+        process
+      )
+      try {
+        if (!step?.action.fromChainId) {
+          throw new Error('Signer is not defined.')
+        }
+
+        const response = await fetchTxErrorDetails(
+          e.transactionHash,
+          step?.action.fromChainId
+        )
+
+        const errorMessage = response?.error_message ?? defaultErrorMessage
+
+        if (errorMessage?.includes(EthersErrorMessage.ERC20Allowance)) {
+          return new TransactionError(
+            LifiErrorCode.AllowanceRequired,
+            e.reason,
+            errorMessage,
+            e.stack
+          )
+        }
+
+        // Error messages other than allowance error will be handled in catch block
+        throw new Error(e)
```

Comment:
Can we return the below `ProviderError` here instead of throwing a new one and then catching it?

Reply:
In the catch block, we are already throwing `ProviderError`. Hence just propagating regular error here.

Reply:
I mean why do we need `try...catch` here in the first place? 🙂

Reply:
As per our previous discussions, to cover for any instance where tenderly API call fails.

If Tenderly API call fails, then resort to our default error message.

Reply:
Ah, okay, that makes sense.